modules/stochastic_tools/src/reporters/AdaptiveMonteCarloDecision.C

```diff
@@ -90,25 +103,45 @@ AdaptiveMonteCarloDecision::execute()
   if (_ais)
   {
     const Real tmp = _ais->getUseAbsoluteValue() ? std::abs(_output_value[0]) : _output_value[0];
-    const bool output_limit_reached = tmp >= _output_limit;
+    bool output_limit_reached;
+
+    /* Checking whether a GP surrogate is used. If it is used, importance sampling is not performed
+    during the training phase of the GP and all proposed samples are accepted until the training
+    phase is completed. Once the training is completed, the importance sampling starts.
+
+    If a GP surrogate is not used, the standard proposal and acceptance/rejection is performed as
+    part of the importance sampling. */
+    bool restart_gp = 0;
+    output_limit_reached = tmp >= _output_limit;
+    if (isParamValid("gp_decision"))
+    {
+      const int training_samples =
```

Comment:
Doing this in execute might be pretty slow. I recommend creating a link to `ActiveLearningGPDecision` in either the constructor or in initial setup.

Reply:
This involves a warehouse lookup which is rather slow, create a link (pointer) in the constructor or in `initialSetup/initialize`.

Reply:
I did this `_gp_used = isParamValid("gp_decision") ? true : false;` in the constructor now. This is not dynamic in that gp_used is either true or false throughout the simulation.

Reply:
I am no longer using `isParamValid` in the execute function anymore.